### PR TITLE
Fix false-positive D405 recognition

### PR DESCRIPTION
**Pull request by shacharoo**
Solves #311.

The problem was that the regex split words when reached an underscore, so "arguments_bla" was caught as "arguments".

### CODE REVIEW

File: src/pydocstyle/checker.py [hunk 1]
```diff
@@ -454,30 +454,42 @@ def _is_a_docstring_section(context):
...
+            * There's no suffix to the section name or it's just a colon AND
+            * The previous line is empty OR it ends with punctuation.
 
         If one of the conditions is true, we will consider the line as
         a section name.
         """
-        section_name_suffix = context.line.lstrip(context.section_name).strip()
+        section_name_suffix = context.line.strip().lstrip(
+            context.section_name.strip()
+        ).strip()
```
Comment (Nurdok) [inline src/pydocstyle/checker.py]: Weird formatting, can you make it look better?

File: src/pydocstyle/checker.py [hunk 2]
```diff
@@ -454,30 +454,42 @@ def _is_a_docstring_section(context):
...
+        section_suffix_is_only_colon = section_name_suffix == ':'
 
         punctuation = [',', ';', '.', '-', '\\', '/', ']', '}', ')']
         prev_line_ends_with_punctuation = \
             any(context.previous_line.strip().endswith(x) for x in punctuation)
 
-        return (is_blank(section_name_suffix) or
-                prev_line_ends_with_punctuation or
-                is_blank(context.previous_line))
+        return ((is_blank(section_name_suffix) or
```
Comment (Nurdok) [inline src/pydocstyle/checker.py]: If we're going verbose 😄:
```.py
this_line_looks_like_a_section_name = is_blank(section_name_suffix) or prev_line_ends_with_punctuation
prev_line_looks_like_end_of_paragraph = prev_line_ends_with_punctuation or is_blank(context.previous_line)
return this_line_looks_like_a_section_name and prev_line_looks_like_end_of_paragraph
```

File: src/tests/test_cases/sections.py [hunk 3]
```diff
@@ -215,3 +215,28 @@ def multiple_sections():
...
     """
+
+
+@expect(_D213)
+def false_positive_section_prefix():
+    """Toggle the gizmo.
+
+    Parameters
+    ----------
+    attributes_are_fun: attributes for the function.
```
Comment (Nurdok) [inline src/tests/test_cases/sections.py]: Perhaps also add a case where the parameter name is _exactly_ a section header name (e.g., `attributes`)?

Comment (shacharoo) [reply src/tests/test_cases/sections.py]: Look at the test below it - it's exactly that.

File: src/pydocstyle/checker.py (hunk 2 above)
Comment (shacharoo) [reply src/pydocstyle/checker.py]: I don't know if you're sarcastic, but I actually think it's better!

**Pull request merged**